solutions/platinum/hr-SubtreesPaths.mdx

```diff
@@ -0,0 +1,191 @@
+---
+id: hr-SubtreesPaths
+source: HR
+title: Subtrees & Paths
+author: Rameez Parwez
+---
+
+[Official Editorial](https://www.hackerrank.com/challenges/subtrees-and-paths/editorial)
+
+## Intuition
+
+The problem requires us to perform two operations: adding a value $x$ to all nodes in the subtree rooted at $t$, and finding the maximum value on the path from $a$ to $b$. From the [Tutorial](https://usaco.guide/plat/hld?lang=cpp#tutorial) we know that Heavy-Light Decomposition(HLD) handles both operations efficiently. Therefore we will use HLD along with a Lazy Segment Tree for the query operations.
+
+The explanation is similar to the one mentioned [here](https://usaco.guide/plat/hld?lang=cpp#template).
+
+## Implementation
+**Time Complexity:** $\mathcal{O} ((N + Q) \log^2(N))$
+<LanguageSection>
+<CPPSection>
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+const int MAX_N = 1e5 + 1, LG = 18, INF = 1e9 + 9;
+// BeginCodeSnip{Lazy Segment Tree}
+struct LST {
+
+	int t[4 * MAX_N], lazy[4 * MAX_N];
+	LST() {
+		fill(t, t + 4 * MAX_N, -INF);
+		fill(lazy, lazy + 4 * MAX_N, 0);
+	}
+
+	inline void push(int n, int b, int e) {
+		if (lazy[n] == 0) { return; }
+		t[n] = t[n] + lazy[n];
+		if (b != e) {
+			lazy[2 * n] = lazy[2 * n] + lazy[n];
+			lazy[2 * n + 1] = lazy[2 * n + 1] + lazy[n];
+		}
+		lazy[n] = 0;
+	}
+
+	inline int combine(int a, int b) { return max(a, b); }
+
+	inline void pull(int n) { t[n] = max(t[2 * n], t[2 * n + 1]); }
+
+	void build(int n, int b, int e) {
+		if (b == e) {
+			t[n] = 0;
+			return;
+		}
+
+		int mid = (b + e) >> 1;
+		build(2 * n, b, mid);
+		build(2 * n + 1, mid + 1, e);
+		pull(n);
+	}
+
+	void upd(int n, int b, int e, int i, int j, int v) {
+		push(n, b, e);
+		if (j < b or e < i) { return; }
+
+		if (i <= b and e <= j) {
+			lazy[n] += v;
+			push(n, b, e);
+			return;
+		}
+
+		int mid = (b + e) >> 1;
+		upd(2 * n, b, mid, i, j, v);
+		upd(2 * n + 1, mid + 1, e, i, j, v);
+		pull(n);
+	}
+
+	int query(int n, int b, int e, int i, int j) {
+		push(n, b, e);
+		if (i > e or b > j) { return -INF; }
```

Comment:
i think most of our code uses `||` and not `or` lol, same goes for `&&` vs `and`

Reply:
changed.